pkg/readers/adapters/amqp.go

```diff
@@ -0,0 +1,50 @@
+package adapters
+
+import (
+	"github.com/softonic/homing-pigeon/pkg/messages"
+	amqpAdapter "github.com/softonic/homing-pigeon/pkg/readers/adapters/amqp"
+	"github.com/streadway/amqp"
+	"log"
+)
+
+type Amqp struct {
+	ConsumedMessages <-chan amqp.Delivery
+	Conn             amqpAdapter.Connection
+	Ch               amqpAdapter.Channel
+}
+
+func (a *Amqp) Listen(writeChannel *chan messages.Message) {
```

Comment:
Would be nice to restrict the channels for their usage to only read and only write due to we do not have bidirectional communication in them.

Example
```
func ping(pings chan<- string, msg string) {     pings <- msg }
--
func pong(pings <-chan string, pongs chan<- string) {     msg := <-pings     pongs <- msg }
```